scripts/checkpoint_converters/convert_griffin_nemo_to_hf.py

```diff
@@ -0,0 +1,148 @@
+import os
+from argparse import ArgumentParser
+
+from omegaconf.omegaconf import OmegaConf
+from transformers import AutoModelForCausalLM
+
+from nemo.collections.nlp.models.language_modeling.megatron_griffin_model import MegatronGriffinModel
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronLMPPTrainerBuilder
+from nemo.collections.nlp.parts.utils_funcs import torch_dtype_from_precision
+from nemo.utils import logging
+
+
+def get_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--hparams_file",
+        type=str,
+        default=f"{os.path.dirname(__file__)}/../../examples/nlp/language_modeling/conf/megatron_griffin_config.yaml",
+        required=False,
+        help="Path config for restoring. It's created during training and may need to be modified during restore if restore environment is different than training. Ex: /raid/nemo_experiments/megatron_gpt/hparams.yaml",
+    )
+    parser.add_argument("--output_path", type=str, default=None, required=True, help="Path to output .nemo file.")
+    parser.add_argument("--input_path", type=str, default=None, required=True)
+    parser.add_argument(
+        "--precision", type=str, default="32", choices=["bf16", "32"], help="Precision for checkpoint weights saved"
+    )
+    args = parser.parse_args()
+    return args
+
+
+def convert(args):
+
+    nemo_config = OmegaConf.load(args.hparams_file)
+    nemo_config.trainer["precision"] = args.precision
+
+    logging.info(f"Loading checkpoint from NeMo: `{args.input_path}`")
+
+    trainer = MegatronLMPPTrainerBuilder(nemo_config).create_trainer()
+
+    nemo_model = MegatronGriffinModel.restore_from(args.input_path, trainer=trainer)
+
+    hf_model = AutoModelForCausalLM.from_pretrained("google/recurrentgemma-2b")
```

Comment:
if we don't have head supprot, shall we do https://github.com/huggingface/transformers/blob/main/src/transformers/models/recurrent_gemma/modeling_recurrent_gemma.py#L639 instead of https://github.com/huggingface/transformers/blob/main/src/transformers/models/recurrent_gemma/modeling_recurrent_gemma.py#L784
Also maybe don't use automodal, but use recurrent gemma directly?

Reply:
Resolved